File: src/main/java/com/hjj/apiserver/domain/BaseEntity.java
package com.hjj.apiserver.domain;

import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.LastModifiedBy;

import javax.persistence.Column;
import javax.persistence.MappedSuperclass;

@MappedSuperclass
@Getter
@Setter
public abstract class BaseEntity extends BaseTimeEntity {

    @Column(columnDefinition = "char(1) default 'N'", nullable = false, insertable = false)
    private char deleteYn;

    @CreatedBy
    @Column(updatable = false)
    private Long createdBy;

    @Column
    @LastModifiedBy
    private Long lastModifiedBy;

}

File: src/main/kotlin/com/hjj/apiserver/domain/BaseEntity.kt
package com.hjj.apiserver.domain

import jakarta.persistence.Column
import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedBy
import org.springframework.data.annotation.LastModifiedBy
import org.springframework.data.jpa.domain.support.AuditingEntityListener

@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
abstract class BaseEntity(
    @Column(columnDefinition = "boolean default false", nullable = false, insertable = false)
    var isDelete: Boolean = false,
    @CreatedBy
    @Column(updatable = false)
    var createdBy: Long? = null,
    @LastModifiedBy
    @Column
    var lastModifiedBy: Long? = null,
) : BaseTimeEntity() {
    fun delete() {
        isDelete = true
    }
}
